graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/builtins/SetPrototypeBuiltins.java

```diff
@@ -229,6 +264,406 @@ protected boolean hasNoObject(Object thisObj, Object key) {
         }
     }
 
+    /**
+     * Implementation of the Set.prototype.intersection().
+     */
+    protected abstract static class JSSetNewOperation extends JSSetOperation {
+
+        @Child protected GetIteratorNode getIteratorNode;
+        @Child protected IteratorStepNode iteratorStepNode;
+        @Child protected IteratorValueNode iteratorValueNode;
+        @Child protected IteratorCloseNode iteratorCloseNode;
+        @Child protected JSFunctionCallNode callFunctionNode;
+        @Child protected PropertyGetNode getAddNode;
+        @Child protected PropertyGetNode getRemoveNode;
+        @Child protected PropertyGetNode getHasNode;
+        protected final BranchProfile iteratorError = BranchProfile.create();
+        protected final BranchProfile adderError = BranchProfile.create();
+
+        protected JSSetNewOperation(JSContext context, JSBuiltin builtin) {
+            super(context, builtin);
+            getIteratorNode = GetIteratorNode.create(context);
+            iteratorStepNode = IteratorStepNode.create(context);
+            iteratorValueNode = IteratorValueNode.create(context);
+            iteratorCloseNode = IteratorCloseNode.create(context);
+        }
+
+        protected Object addEntryFromIterable(Object target, Object iterable, Object adder) {
+            if (!JSRuntime.isCallable(adder)) {
```

Comment:
Instead of `JSRuntime.isCallable`, please use a Node as well:

```
@Child private IsCallableNode isCallableNode;

protected final boolean isCallable(Object callback) {
     if (isCallableNode == null) {
         CompilerDirectives.transferToInterpreterAndInvalidate();
         isCallableNode = insert(IsCallableNode.create());
     }
     return isCallableNode.executeBoolean(callback);
}
```

This `isCallable` you can then reuse in all your nodes.